### ISSUE TITLE
package is not available in a stable-enough version

### ISSUE THREAD

**Issue by taleemhussain** [xvilo/harvest-forecast#3]
I am getting this error while installing these Packages in the larvel.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
@xvilo @samuele-mattiuzzo

**Comment by samuele-mattiuzzo** [comment]
How are you trying to install it? What is the full stacktrace you're getting out of composer? What's your PHP version?

**Comment by xvilo** [comment]
I would like to have more information:
- what is the exact command you're running **and** its output?
- what version of PHP are you using?
- what version of Composer are you using?
- Any other thing we could know about your environment?

**Comment by taleemhussain** [comment]
I am using the following for laravel 8. 
Composer version 2.0.13
PHP 8.
@xvilo @samuele-mattiuzzo

**Comment by taleemhussain** [comment]
@xvilo @samuele-mattiuzzo 
Any suggestion?

**Comment by samuele-mattiuzzo** [comment]
I'm on the same versions that you're using, and it's not giving me any problems, so unless you post the full stack trace of the error and more info, I'm afraid I cannot assist you

**Comment by taleemhussain** [comment]
I am using with Laravel is there any limitation with that?

**Comment by tomjn** [comment]
I'm getting this, I created a new folder, ran `git init` then copy pasted `composer require xvilo/harvest-forecast` and got this:

```sh
❯ composer require xvilo/harvest-forecast
Using version ^1.1 for xvilo/harvest-forecast
./composer.json has been created
Running composer update xvilo/harvest-forecast
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - xvilo/harvest-forecast[1.1.0, ..., v1.1.1] require php-http/client-implementation ^1.0 -> could not be found in any version, but the following packages provide it:
      - symfony/http-client Provides powerful methods to fetch HTTP resources synchronously or asynchronously
      - php-http/guzzle6-adapter Guzzle 6 HTTP Adapter
      - symfony/symfony The Symfony PHP framework
      - php-http/curl-client PSR-18 and HTTPlug Async client with cURL
      - kriswallsmith/buzz Lightweight HTTP client
      - php-http/guzzle7-adapter Guzzle 7 HTTP Adapter
      - php-http/mock-client Mock HTTP client
      - php-http/socket-client Socket client for PHP-HTTP
      - php-http/guzzle5-adapter Guzzle 5 HTTP Adapter
      - php-http/react-adapter React HTTP Adapter
      - voku/httpful A Readable, Chainable, REST friendly, PHP HTTP Client
      - jorge-matricali/http-client A wrapper of libcurl that implements PSR-7 HTTP message interface.
      - php-http/cakephp-adapter Cake adapter for PHP-HTTP
      - php-http/buzz-adapter Buzz HTTP Adapter
      - code-tool/curl-client cURL client
      - windwalker/http Windwalker Http package
      - phpgt/fetch Asynchronous HTTP client with promises for PHP 7 applications.
      - mushroomcloud/guzzle6-adapter Guzzle 6 HTTP Adapter
      - windwalker/framework The next generation PHP framework.
      - swisnl/php-http-fixture-client Fixture client for PHP-HTTP
      ... and 33 more.
      Consider requiring one of these to satisfy the php-http/client-implementation requirement.
    - Root composer.json requires xvilo/harvest-forecast ^1.1 -> satisfiable by xvilo/harvest-forecast[1.1.0, v1.1.1].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.

Installation failed, deleting ./composer.json.
```

> what is the exact command you're running and its output?
```sh
mkdir test
cd test
git init
composer require xvilo/harvest-forecast
```

> what version of PHP are you using?

```sh
❯ php --version
PHP 7.4.25 (cli) (built: Oct 23 2021 18:05:05) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.25, Copyright (c), by Zend Technologies
```

> what version of Composer are you using?

```sh
❯ composer --version
Composer version 2.1.8 2021-09-15 13:55:14
```

> Any other thing we could know about your environment?

MacOS 11 Big Sur, PHP installed via homebrew

@xvilo I believe the problem is simple. This package requires a HTTP client, but never actually specifies a package to satisfy it, just that it needs satisfying. The usage instructions then neglect to mention this, leading to confusion when someone literally follows exactly what you said they should do and get unexpected errors.

As proof, I tries this command which works fine:

```sh
composer require xvilo/harvest-forecast php-http/guzzle6-adapter
```

TLDR: The problem is incorrect/improper documentation in the readme, and missing information

**Comment by tomjn** [comment]
#5 should resolve this and some other things

**Comment by xvilo** [comment]
Yep, you're right. You're open to use your own PSR HTTP Client implementation. This _could_ be Guzzle6 or the Symfony HTTP Client Component. 

I had chosen to just only install the library and, in case one has no PSR HTTP Client implementation add it the error to the FAQ which could be confusing for developers who have not encountered this before. 

@taleemhussain I am closing this issue as you have not provided any additional and clear information and I think it might be fixed with the additions in the README.md file.
